graphbolt/src/csc_sampling_graph.cc

```diff
@@ -256,8 +257,76 @@ inline torch::Tensor UniformPick(
       picked_neighbors =
           torch::randint(offset, offset + num_neighbors, {fanout}, options);
     } else {
-      picked_neighbors = torch::randperm(num_neighbors, options);
-      picked_neighbors = picked_neighbors.slice(0, 0, fanout) + offset;
+      picked_neighbors = torch::empty({fanout}, options);
+      AT_DISPATCH_INTEGRAL_TYPES(
+          picked_neighbors.scalar_type(), "UniformPick", ([&] {
+            scalar_t* picked_neighbors_data =
+                picked_neighbors.data_ptr<scalar_t>();
+            if (fanout < num_neighbors / 10) {
```

Comment:
// We use different sampling strategies for different sampling intensities.
if (fanout >= num_neighbors / 10) {
  // Move comment L307 - L317 here.
  code.
} else if (fanout < 64) {
  // Move L266-L267 here.
  code.
} else {
  // ...
}

Reply:
Got